src/scenic/core/requirements.py

```diff
@@ -130,18 +151,23 @@ def __str__(self):
             return f'"{self.ty.value}" on line {self.line}'
 
 class BoundRequirement:
-    def __init__(self, compiledReq, sample):
+    def __init__(self, compiledReq, sample, proposition):
         self.ty = compiledReq.ty
         self.closure = compiledReq.closure
         self.line = compiledReq.line
         self.name = compiledReq.name
-        assert compiledReq.prob == 1
         self.sample = sample
+        self.compiledReq = compiledReq
+        self.proposition = proposition
 
     def isTrue(self):
         return self.value()
 
     def value(self):
+        one_time_monitor = self.proposition.create_monitor()
```

Comment:
Is this necessary? If you don't pass a monitor at all, won't the closure test the correct proposition anyway? If so, there shouldn't be any need for 3 different methods to check the truth value (even for the requirements with built-in monitors, they're confusing: if we need them, let's put docstrings explaining the difference).